components/onboarding/Onboarding.tsx

```diff
@@ -0,0 +1,163 @@
+'use client';
+import { useSearchParams, useRouter } from 'next/navigation';
+import React, { useState } from 'react';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import toast from 'react-hot-toast';
+
+import { User } from '@prisma/client';
+import { updateUser } from '@/lib/actions/user.actions';
+import revalidateSession from '@/lib/actions/auth.actions';
+import {
+  IOnboardingSchema,
+  OnboardingSchema,
+} from '@/lib/validations/UserSchema';
+
+import BasicInformation from '@/components/onboarding/BasicInformation';
+import LearningGoals from '@/components/onboarding/LearningGoals';
+import KnowledgeLevel from '@/components/onboarding/KnowledgeLevel';
+import Availability from '@/components/onboarding/Availability';
+import OnboardingVisualStepper from '@/components/onboarding/OnboardingVisualStepper';
+import Button from '@/components/shared/ui/Button';
+
+const Onboarding = ({ user }: { user: User }) => {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const stepFromParams = parseInt(searchParams.get('step') ?? '1', 10);
+
+  const [step, setStep] = useState(stepFromParams);
+  let fields = [] as Partial<keyof IOnboardingSchema>[];
+
+  const useFormHelpers = useForm<IOnboardingSchema>({
+    defaultValues: {
+      name: user?.name ?? '',
+      image: user?.image ?? '',
+      location: '',
+      portfolio: '',
+      goals: [],
+      knowledgeLevel: [],
+      techStack: [],
+      availability: false,
+      startDate: new Date(),
+      endDate: new Date(),
+    },
+    resolver: zodResolver(OnboardingSchema),
+  });
+
+  const { watch, handleSubmit, trigger } = useFormHelpers;
+  const formData = watch();
+
+  const filterData = (data: IOnboardingSchema) => {
+    const filteredData = Object.keys(data).filter((key) =>
+      fields.includes(key as keyof IOnboardingSchema)
+    );
+    const dataToSend = filteredData.reduce((acc, cur) => {
+      return {
+        ...acc,
+        [cur]: data[cur],
```

Comment:
```suggestion
        [cur]: data[cur as keyof IOnboardingSchema],
```
will fix your typescript error here.